nix/modules/common.nix

```diff
@@ -0,0 +1,93 @@
+# SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+{ lib, ... }:
+
+with lib;
+rec {
+  sharedOptions = {
+
+    logVerbosity = mkOption {
+      type = types.enum [ "fatal" "error" "warning" "notice" "info" "debug" ];
+      default = "warning";
+      description = ''
+        Level of logs verbosity. Possible values are:
+        fatal, error, warn, notice, info or debug.
+      '';
+    };
+
+  };
+
+  daemonOptions = sharedOptions // {
+
+    baseProtocol = mkOption {
+      type = types.enum [ "007-PsDELPH1" "008-PtEdo2Zk" "009-PsFLoren"];
+      description = ''
+        Base protocol version,
+        only '007-PsDELPH1', '008-PtEdo2Zk', and '009-PsFLoren' are supported.
+      '';
+      example = "008-PtEdo2Zk";
+    };
+
+    rpcPort = mkOption {
+      type = types.int;
+      default = 8732;
+      example = 8732;
+      description = ''
+        Tezos node RPC port.
+      '';
+    };
+
+  };
+
+  genDaemonConfig = instancesCfg: service-name: service-pkgs: service-script:
+    mkIf (instancesCfg != {}) {
+      users = mkMerge (flip mapAttrsToList instancesCfg (node-name: node-cfg: genUsers node-name ));
+      systemd = mkMerge (flip mapAttrsToList instancesCfg (node-name: node-cfg:
+        let tezos-service = service-pkgs."${node-cfg.baseProtocol}";
+        in {
+          services."tezos-${node-name}-tezos-${service-name}" = genSystemdService node-name node-cfg service-name // rec {
+            bindsTo = [ "network.target" "tezos-${node-name}-tezos-node.service" ];
+            after = bindsTo;
+            preStart =
+              ''
+                sleep 10 # to allow tezos-node.service to successfully start
```

Comment:
Looks suspicious, 10s can be not enough in case the node is generating an identity, let's use curl to check whether the node is up, see [this](https://gitlab.com/morley-framework/local-chain/-/blob/master/run-local-chain.sh#L54)